### ISSUE TITLE
How to mock and test request with content passed as byte iterator

### ISSUE THREAD

**Issue by adamko147** [Colin-b/pytest_httpx#148]
Hello team, 

I have `httpx` code that passes `content` to request using byte iterator. When trying the test code using `pytest_httpx` and `match_content` on mocked request, I get assertion error and the `aiter_bytes` is never called. Here's what I'm trying...

```python
import httpx
import pytest


async def aiter_bytes():
    yield b"h"
    yield b"e"
    yield b"l"
    yield b"l"
    yield b"o"


@pytest.mark.asyncio
async def test_httpx_get(httpx_mock):
    httpx_mock.add_response(method="PUT", url="http://put", content=b"world", match_content=b"hello")
    async with httpx.AsyncClient() as client:
        res = await client.put("http://put", content=aiter_bytes())
        assert res.text == "world"
```

and the exception when runnning the test
```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Request('PUT', 'http://put')>

    def read(self) -> bytes:
        """
        Read and return the request content.
        """
        if not hasattr(self, "_content"):
>           assert isinstance(self.stream, typing.Iterable)
E           AssertionError
```

When I remove the `match_content` parameter, the test passes, but the `aiter_bytes` is never called. Is there anything I'm missing? I'm not sure whether this is `pytest_httpx` or `httpx` issue, although when I run the code without `pytest` in real program, it works and sends the data properly.

Thank you,
Adam

**Comment by Colin-b** [comment]
Hello,

[The documentation is already explaining how to stream chunks](https://github.com/Colin-b/pytest_httpx?tab=readme-ov-file#reply-by-streaming-chunks)

If something is not working in the provided example, feel free to report it.

Thanks

**Comment by adamko147** [comment]
> Hello,
> 
> [The documentation is already explaining how to stream chunks](https://github.com/Colin-b/pytest_httpx?tab=readme-ov-file#reply-by-streaming-chunks)
> 
> If something is not working in the provided example, feel free to report it.
> 
> Thanks

Hi @Colin-b, thanks for reaching out. I'am aware of the documentation, although the docs covers the case when mocked url produce the stream, which is not wham I'm trying to achieve.

Following code passes fine
```
async def test_httpx_put(httpx_mock):
    httpx_mock.add_response(method="PUT", url="http://put", content=b"world", match_content=b"hello")
    async with httpx.AsyncClient() as client:
        res = await client.put("http://put", content=b"hello")
        assert res.text == "world"
```
and checks that I call http://put with "hello" content.

when I change the implementation of sending the data to async iterator as follows
```python
async def aiter_bytes():
    yield b"h"
    yield b"e"
    yield b"l"
    yield b"l"
    yield b"o"


@pytest.mark.asyncio
async def test_httpx_put_iter(httpx_mock):
    httpx_mock.add_response(method="PUT", url="http://put", content=b"world", match_content=b"hello")
    async with httpx.AsyncClient() as client:
        res = await client.put("http://put", content=aiter_bytes())
        assert res.text == "world"
```
the execution fails with error 
```
self = <Request('PUT', 'http://put')>

    def read(self) -> bytes:
        """
        Read and return the request content.
        """
        if not hasattr(self, "_content"):
>           assert isinstance(self.stream, typing.Iterable)
E           AssertionError
```

The code should work, based on httpx documentation you can pass content as _Binary content to include in the body of the request, as bytes or a byte iterator._

**Comment by Colin-b** [comment]
My bad, I read this too fast. Indeed this looks like a bug !

Thanks for reporting, I'll try to find some time

**Comment by Colin-b** [comment]
Fixed in version 0.31.1 available on pypi
